code/modules/modular_computers/file_system/programs/ntdownloader.dm

```diff
@@ -100,6 +100,8 @@
 		if(NTNET_ETHERNET_SIGNAL)
 			download_netspeed = NTNETSPEED_ETHERNET
 	if(download_netspeed)
+		if(HAS_TRAIT(computer, TRAIT_MODPC_HALVED_DOWNLOAD_SPEED))
+			download_completion *= 0.5
```

Comment:
this seems like it should be modifying download_netspeed rather than the completion?
```suggestion
		if(HAS_TRAIT(computer, TRAIT_MODPC_HALVED_DOWNLOAD_SPEED))
			download_netspeed *= 0.5
```